frontend/src/metabase-types/analytics/embed-share.ts

```diff
@@ -10,15 +10,24 @@ type ValidateEvent<
     Record<Exclude<keyof T, keyof EmbedShareEventSchema>, never>,
 > = T;
 
+type EmbeddingEventName =
+  | "embedding"
+  | "sdk_embedding"
+  | "interactive_embedding"
+  | "static_embedding";
+
+type EmbeddingEventEnabled = `${EmbeddingEventName}_enabled`;
+type EmbeddingEventDisabled = `${EmbeddingEventName}_disabled`;
```

Comment:
I start to see things much clearer once I'm not too sleepy 😆 . This can be done later, but I think we should refrain from using dynamic strings as much as possible. This makes it hard to search, imagine you want to know where the type of `interactive_embedding_enabled` is, the only result that is returned would be the value in the snowplow schema.

We encountered some of the dynamic strings during the CSS migration which made our style break because we couldn't find the usage at first, and I don't want us to repeat that again.